controllers/operatorpolicy_controller.go

```diff
@@ -332,6 +346,74 @@ func (r *OperatorPolicyReconciler) buildResources(policy *policyv1beta1.Operator
 	return sub, opGroup, updateStatus(policy, validationCond(validationErrors)), nil
 }
 
+// applySubscriptionDefaults will set the subscription channel, source, and sourceNamespace when they are unset by
+// utilizing the PackageManifest API.
+func (r *OperatorPolicyReconciler) applySubscriptionDefaults(
+	ctx context.Context, subscription *operatorv1alpha1.Subscription,
+) error {
+	subSpec := subscription.Spec
+
+	defaultsNeeded := subSpec.Channel == "" || subSpec.CatalogSource == "" || subSpec.CatalogSourceNamespace == ""
+
+	if !defaultsNeeded {
+		return nil
+	}
+
+	// PackageManifests come from an API server and not a Kubernetes resource, so the DynamicWatcher can't be used since
+	// it utilizes watches. The namespace doesn't have any meaning but is required.
+	packageManifest, err := r.DynamicClient.Resource(packageManifestGVR).Namespace("default").Get(
+		ctx, subSpec.Package, metav1.GetOptions{},
+	)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return errors.New(
+				"the subscription defaults could not be determined because the PackageManifest was not found",
+			)
+		}
+
+		log.Error(err, "Failed to get the PackageManifest", "name", subSpec.Package)
+
+		return errors.New(
+			"the subscription defaults could not be determined because the PackageManifest API returned an error",
+		)
+	}
+
+	catalog, _, _ := unstructured.NestedString(packageManifest.Object, "status", "catalogSource")
+	catalogNamespace, _, _ := unstructured.NestedString(packageManifest.Object, "status", "catalogSourceNamespace")
+
+	if catalog == "" || catalogNamespace == "" {
+		return errors.New(
+			"the subscription defaults could not be determined because the PackageManifest didn't specify a catalog",
+		)
+	}
+
+	if (subSpec.CatalogSource != "" && subSpec.CatalogSource != catalog) ||
+		(subSpec.CatalogSourceNamespace != "" && subSpec.CatalogSourceNamespace != catalogNamespace) {
+		return errors.New(
+			"the subscription defaults could not be determined because the catalog doesn't match those in the " +
+				"PackageManifest",
```

Comment:
```suggestion
			"the subscription defaults could not be determined because the catalog specified in the " +
				"policy does not match what was found in the PackageManifest on the cluster",
```

Especially if the user is not familiar with OLM, I think we should give some extra hints on where these values came from.